README.md

```diff
@@ -1,39 +1,601 @@
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_avro('path/to/avro/file')
+    ```
+
+### CSV Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from a **.csv** or **.csv.gz** file.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/CSV)
+- **Usage**:
+    ```ruby
+    #! Partially require just CSV Importer
+    require 'daru/io/importers/csv'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::CSV.new('path/to/csv.gz/file', skiprows: 10, compression: :gzip).call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_csv('path/to/csv.gz/file', skiprows: 10, compression: :gzip)
+    ```
+
+### Excel Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from a **.xls** file.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/Excel)
+- **Gem Dependencies**: `spreadsheet` gem
+- **Usage**:
+    ```ruby
+    #! Partially require just Excel Importer
+    require 'daru/io/importers/excel'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::Excel.new('path/to/xls/file', worksheet_id: 1).call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_excel('path/to/xls/file', worksheet_id: 1)
+    ```
+
+### Excelx Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from a **.xlsx** file.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/Excelx)
+- **Gem Dependencies**: `roo` gem
+- **Usage**:
+    ```ruby
+    #! Partially require just Excel Importer
+    require 'daru/io/importers/excelx'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::Excelx.new('path/to/xlsx/file', sheet: 2, skiprows: 10, skipcols: 2).call
+
+    #! Usage from Daru::DataFrame
+    require 'daru/io/importers/excel'
+    Daru::DataFrame.from_excel('path/to/xlsx/file', sheet: 2, skiprows: 10, skipcols: 2)
+    ```
+
+### HTML Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+**Note: This module works only for static tables on a HTML page, and won't work in cases where the table is being loaded into the HTML table by inline Javascript. This is how the Mechanize gem works, and the HTML Importer also follows suit.**
+
+Imports an **Array** of **Daru::DataFrame**s from a **.html** file or website.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/HTML)
+- **Gem Dependencies**: `mechanize` gem
+- **Usage**:
+    ```ruby
+    #! Partially require just HTML Importer
+    require 'daru/io/importers/html'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::HTML.new('website/domain/url', match: 'market', name: 'Shares analysis').call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_html('website/domain/url', match: 'market', name: 'Shares analysis')
```

Comment:
I'd say
```ruby
# local
Daru::DataFrame.from_html('path/to/file.html', match: 'market', name: 'Shares analysis')
# or remote
Daru::DataFrame.from_html('https://some/url/with/tables', match: 'market', name: 'Shares analysis')
```
...this showcase is MUCH more useful than adding obvious `Daru::IO::Importers::HTML.new().call` to each importer, which I am against.